File: vaporBackEnd/Sources/App/Models/Player.swift
//
//  File.swift
//  
//
//  Created by Benny De Bock on 18/05/2022.
//

import Foundation
import Vapor

struct PlayerDB: Content {
    
}

struct Player: Content {
    let Id: UUID
    let PlayerID: Int
    let Name: String
    let Gender: Bool
    let Active: Bool
    let Birthyear: Int?
    let PlayHand: String?
    let PlayStyle: String?
    let Grip: String?
    let Country: Country?
}

struct Country: Content {
    let Id: UUID
    let Code: String
}

struct PlayerDictionary: Content {
    let dict: [String: Double]
}

File: nodeBackEnd/Models/player.ts
// interface Player {
//     id: String,
//     playerId: number,
//     name: String,
//     gender: boolean,
//     active: boolean,
//     birthyear?: number
//     playHand?: String,
//     playStyle?: String,
//     grip?: String,
//     country?: Country
// }

// interface Country {
//     id: String,
//     code: String
// }

export declare type Player = {
    id?: string,
    playerId: number,
    name: string,
    gender: boolean,
    active: boolean,
    birthyear?: number
    playHand?: string,
    playStyle?: string,
    grip?: string,
    country?: Country
}

export declare type Country = {
    id: string,
    code?: string
}